### DISCUSSION TITLE
EG 4.0 Release

### DISCUSSION THREAD

**Discussion by waaffles** [jupyter/enterprise_gateway#1391] [General]
TL;DR: Any rough estimate as to when EG 4.0 will release?

Hey team! We're looking at investing time into integrating EG into our current architecture and are looking specifically at how we can leverage both local and remote kernels. I had started a discussion [here](https://discourse.jupyter.org/t/can-jupyterlab-have-both-local-and-remote-kernels/28916) in the jupyter forums. Kernel and Gateway Provisioners seem to be more in line with what we're talking about but we also want multi user and cluster support. We understand EG 4.0 will include support for kernel Provisioners as well. We're wondering if we'll need to spend time implementing our own process proxy or should instead focus on kernel Provisioners. It depends on when we expect EG 4.0 to be released so I'm wondering if we have any sort of rough timeline or what things are blocking or if it's on indefinite hold, etc?
